packages/smooth_app/lib/pages/product/new_product_page.dart

```diff
@@ -0,0 +1,271 @@
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/model/Attribute.dart';
+import 'package:openfoodfacts/model/Product.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/cards/data_cards/image_upload_card.dart';
+import 'package:smooth_app/cards/data_cards/score_attribute_card.dart';
+import 'package:smooth_app/cards/expandables/attribute_list_expandable.dart';
+import 'package:smooth_app/data_models/product_preferences.dart';
+import 'package:smooth_app/database/dao_product_extra.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/helpers/attributes_card_helper.dart';
+import 'package:smooth_app/helpers/launch_url_helper.dart';
+import 'package:smooth_app/pages/product/common/product_dialog_helper.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_ui_library/widgets/smooth_card.dart';
+
+class NewProductPage extends StatefulWidget {
+  const NewProductPage({
+    required this.product,
```

Comment:
nit: Should this be a nameless positional parameter instead, since this will likely be called as `NewProductPage(product)`?

Reply:
Nameless parameter SG